File: src/JavaScript/Piece of Cake!/PieceOfCake2.js
const readline = require('readline');
const rl = readline.createInterface(process.stdin, process.stdout);

rl.on('line', line => {
    [n, h, v] = line.split(' ');
    let maxh = Math.max(h, n - h);
    let maxv = Math.max(v, n - v);
    console.log(maxh * maxv * 4);
    rl.close();
});

File: src/Python 3/Piece of Cake!/PieceOfCake2.py
n, h, v = input().split()
n = int(n)
h = int(h)
v = int(v)
horLarger = 0
verLarger = 0
horLarger = max(h, n - h)
verLarger = max(v, n - v)
print(horLarger * verLarger * 4)